source/opt/workaround_image_operands.cpp

```diff
@@ -0,0 +1,173 @@
+                              sampled_image_usage->opcode() ==
+                                  SpvOpImageSampleDrefExplicitLod ||
+                              sampled_image_usage->opcode() ==
+                                  SpvOpImageSampleProjDrefImplicitLod ||
+                              sampled_image_usage->opcode() ==
+                                  SpvOpImageSampleProjDrefExplicitLod ||
+                              sampled_image_usage->opcode() ==
+                                  SpvOpImageSparseSampleDrefImplicitLod ||
+                              sampled_image_usage->opcode() ==
+                                  SpvOpImageSparseSampleDrefExplicitLod ||
+                              sampled_image_usage->opcode() ==
+                                  SpvOpImageSparseSampleProjDrefImplicitLod ||
+                              sampled_image_usage->opcode() ==
+                                  SpvOpImageSparseSampleProjDrefExplicitLod ||
+                              sampled_image_usage->opcode() ==
+                                  SpvOpImageSparseDrefGather) {
+                            has_depth = true;
+                          }
+                          // If any of these are a dref lookup
+                          if (!has_depth) {
+                            return;
+                          }
+                          analysis::Type* t = type_mgr->GetType(
+                              type_mgr->GetId(img_type->sampled_type()));
+                          analysis::Image img(
+                              t, img_type->dim(), 1,
+                              img_type->is_arrayed(),
+                              img_type->is_multisampled(), img_type->sampled(),
+                              img_type->format(), img_type->access_qualifier());
+                          // Make a new image type identical to the previous, but 
+                          // mark off the depth bit.
+                          for (auto& dec : img_type->decorations()) {
+                            img.AddDecoration(std::vector<uint32_t>(dec));
+                          }
+                          uint32_t new_image =
+                              type_mgr->GetTypeInstruction(&img);
+
+                          // Create a new OpTypeSampledImage with the given type. 
+                          analysis::SampledImage sampled_img(
+                              type_mgr->GetType(new_image));
+                          for (auto& dec : sampled_image_type->decorations()) {
+                            sampled_img.AddDecoration(
+                                std::vector<uint32_t>(dec));
+                          }
+
+                          uint32_t sampled_image_type =
+                              type_mgr->GetTypeInstruction(&sampled_img);
+
+                          // Create a new OpSampledImage value that uses that
+                          // type.
+                          auto new_sampled_image =
```

Comment:
If I use the def_use_mgr, how do I make sure that these get tracked correctly, I tried to register them, but it didn't seem to work.

Reply:
You need to call `context()->AnalyzeDefUse(new_sampled_image)` to make the def-use manager aware of the new instruction.  Do this after you have finished updating the instruction with the new result id.